# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ntp/gcc-4.0.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- ntp-4.2.0/include/ntp.h.vanilla	2005-09-04 18:53:11.000000000 +0000
+++ ntp-4.2.0/include/ntp.h	2005-09-04 18:54:33.000000000 +0000
@@ -5,6 +5,8 @@
 #define NTP_H
 
 #include "ntp_types.h"
+#include "ntp_fp.h"
+
 #include <math.h>
 #ifdef OPENSSL
 #include "ntp_crypto.h"
--- ntp-4.2.0/include/ntp_stdlib.h.vanilla	2005-09-04 17:02:38.000000000 +0000
+++ ntp-4.2.0/include/ntp_stdlib.h	2005-09-04 18:58:43.000000000 +0000
@@ -7,6 +7,7 @@
 
 #include "ntp_types.h"
 #include "ntp_string.h"
+#include "ntp_refclock.h"
 #include "l_stdlib.h"
 
 /*
@@ -134,7 +135,6 @@
 extern u_int	cache_keylen;		/* key length */
 
 /* clocktypes.c */
-struct clktype;
 extern struct clktype clktypes[];
 
 /* getopt.c */
--- ntp-4.2.0/include/ntp_control.h.vanilla	2005-09-04 18:56:48.000000000 +0000
+++ ntp-4.2.0/include/ntp_control.h	2005-09-04 18:57:17.000000000 +0000
@@ -4,6 +4,9 @@
 
 #include "ntp_types.h"
 
+#ifndef NTP_CONTROL_H
+#define NTP_CONTROL_H
+
 struct ntp_control {
 	u_char li_vn_mode;		/* leap, version, mode */
 	u_char r_m_e_op;		/* response, more, error, opcode */
@@ -281,3 +284,5 @@
 #define	TYPE_SYS	1
 #define	TYPE_PEER	2
 #define	TYPE_CLOCK	3
+
+#endif
--- ntp-4.2.0/include/ntpd.h.vanilla	2005-09-04 18:54:49.000000000 +0000
+++ ntp-4.2.0/include/ntpd.h	2005-09-04 18:55:58.000000000 +0000
@@ -3,8 +3,8 @@
  */
 
 #include "ntp_syslog.h"
-#include "ntp_fp.h"
 #include "ntp.h"
+#include "ntp_control.h"
 #include "ntp_select.h"
 #include "ntp_malloc.h"
 #include "ntp_refclock.h"
@@ -226,7 +226,6 @@
 #endif
 
 /* ntp_control.c */
-struct ctl_trap;
 extern struct ctl_trap ctl_trap[];
 extern int	num_ctl_traps;
 extern keyid_t	ctl_auth_keyid;		/* keyid used for authenticating write requests */
